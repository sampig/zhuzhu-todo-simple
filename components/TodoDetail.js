import React from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, Text, View } from 'react-native'
import i18n from 'util/i18n-utils'
import { convertDateToString } from 'util/date-utils'

const styles = StyleSheet.create({
  container: {
    flex: 1,
    display: 'flex',
    marginLeft: 16,
    marginRight: 16,
    alignItems: 'flex-start',
    justifyContent: 'center',
  },
  title: {
    fontSize: 30,
  },
  createTime: {
    fontSize: 18,
  },
})

/**
 * Component to a to-do detail page.
 * @param {object} route - The route object in react native.
 * @returns {node}       - React node
 * @constructor
 */
function TodoDetail ({ route }) {
  const { todoItem } = route.params
  return todoItem ? (
    <View style={styles.container}>
      <Text style={styles.title}>{todoItem.title}</Text>
      <Text style={styles.createTime}>{i18n.t('todoCreateAt')} {convertDateToString(todoItem.createTime)}</Text>
    </View>
  ) : null
}

TodoDetail.propTypes = {
  route: PropTypes.object,
}

export default TodoDetail
