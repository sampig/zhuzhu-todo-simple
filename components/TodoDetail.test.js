import React from 'react'
import renderer from 'react-test-renderer'
import TodoDetail from './TodoDetail'


describe('The TodoDetail component', () => {
  it('renders without crashing', () => {
    const route = { params: { todoItem: { id: 'testId', title: 'test title', createTime: 1 } } }
    const render = renderer.create(<TodoDetail route={route} />).toJSON()
    expect(render).toMatchSnapshot()
  })

  it('return null when no item in route', () => {
    const route = { params: {} }
    const render = renderer.create(<TodoDetail route={route} />).toJSON()
    expect(render).toMatchSnapshot()
  })
})
