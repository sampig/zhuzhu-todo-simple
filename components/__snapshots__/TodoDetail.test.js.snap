// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The TodoDetail component renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flex": 1,
      "justifyContent": "center",
      "marginLeft": 16,
      "marginRight": 16,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 30,
      }
    }
  >
    test title
  </Text>
  <Text
    style={
      Object {
        "fontSize": 18,
      }
    }
  >
    Create at: 
     
    1970-01-01 00:00:00
  </Text>
</View>
`;

exports[`The TodoDetail component return null when no item in route 1`] = `null`;
